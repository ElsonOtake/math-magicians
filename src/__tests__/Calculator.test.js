import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../components/Calculator.js';

describe('Calculator', () => {
  test('renders Calculator component', () => {
    render(<App />);
    expect(screen.getByRole('table')).toBeInTheDocument();
    expect(screen.getByRole('table')).toBeVisible();
    expect(screen.queryByText(/Todo List React/)).toBeNull();
    expect(screen.getByText(/Let's do some math!/)).toBeTruthy();
    expect(screen.getByRole('heading')).toBeInTheDocument();
    expect(screen.getByRole('heading')).toBeVisible();
    const total = screen.getByRole('columnheader');
    fireEvent.click(screen.getByText('5'));
    expect(screen.debug(total)).toMatchSnapshot();
  });
});