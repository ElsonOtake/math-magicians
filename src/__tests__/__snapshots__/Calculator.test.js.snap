// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator renders Calculator component 1`] = `undefined`;
